src/server/boot.ts

```diff
@@ -1,26 +1,40 @@
-import { colors } from "./deps.ts";
+import { colors, parse } from "./deps.ts";
 import { ServeHandler } from "./types.ts";
 
+const onListen =
+  (opts: Partial<Deno.ServeTlsOptions>) =>
+  (params: { hostname: string; port: number }) => {
+    const address = colors.cyan(
+      `http://${opts.hostname || params.hostname}:${opts.port || params.port}/`,
+    );
+    console.log(
+      "\n",
+      colors.bgRgb8(colors.black(colors.bold(" 🍋 Fresh ready ")), 121),
+      `\n    ${colors.bold("Local:")} ${address}\n\n`,
+    );
+  };
+
 export async function startServer(
   handler: Deno.ServeHandler,
   opts: Partial<Deno.ServeTlsOptions>,
 ) {
-  if (!opts.onListen) {
-    opts.onListen = (params) => {
-      console.log();
-      console.log(
-        colors.bgRgb8(colors.black(colors.bold(" 🍋 Fresh ready ")), 121),
-      );
+  const { host: hostFlag, port: portFlag } = parse(Deno.args, {
+    string: ["host", "port"],
+    alias: { h: "host", p: "port" },
+  });
+  const portEnv = Deno.env.get("PORT");
+  const hostEnv = Deno.env.get("HOST");
 
-      const address = colors.cyan(`http://localhost:${params.port}/`);
-      const localLabel = colors.bold("Local:");
-      console.log(`    ${localLabel} ${address}\n`);
-    };
+  if (portEnv !== undefined || portFlag !== undefined) {
+    opts.port = parseInt((portFlag || portEnv) as string, 10);
```

Comment:
Isn't 10 the default? Do you really need to declare it?

Reply:
It's unfortunately a little more complex. Quoting from [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#description):

> - If the input string, with leading whitespace and possible +/- signs removed, begins with 0x or 0X (a zero, followed by lowercase or uppercase X), radix is assumed to be 16 and the rest of the string is parsed as a hexadecimal number.
> - If the input string begins with any other value, the radix is 10 (decimal).

So better to always pass the radix argument when using `parseInt`

Reply:
Thanks for the link and the clarification!